tests/phpunit/tests/blocks/register.php

```diff
@@ -231,6 +290,115 @@ public function test_wrong_array_index_do_not_register_block_script_handle() {
 		$this->assertFalse( $result );
 	}
 
+	/**
+	 * @ticket 60233
+	 */
+	public function test_field_not_found_register_block_script_module_id() {
+		$result = register_block_script_module_id( array(), 'viewScriptModule' );
+
+		$this->assertFalse( $result );
+	}
+
+	/**
+	 * @ticket 60233
+	 */
+	public function test_empty_string_value_do_not_register_block_script_module_id() {
+		$metadata = array( 'viewScriptModule' => '' );
+		$result   = register_block_script_module_id( $metadata, 'viewScriptModule' );
+
+		$this->assertFalse( $result );
+	}
+
+	/**
+	 * @ticket 60233
+	 */
+	public function test_empty_array_value_do_not_register_block_script_module_id() {
+		$metadata = array( 'viewScriptModule' => array() );
+		$result   = register_block_script_module_id( $metadata, 'viewScriptModule' );
+
+		$this->assertFalse( $result );
+	}
+
+	/**
+	 * @ticket 60233
+	 */
+	public function test_wrong_array_index_do_not_register_block_script_module_id() {
+		$metadata = array( 'viewScriptModule' => array( 'test-module_id' ) );
+		$result   = register_block_script_module_id( $metadata, 'script', 1 );
+
+		$this->assertFalse( $result );
+	}
+
+	/**
+	 * @ticket 60233
+	 */
+	public function test_missing_asset_file_register_block_script_module_id() {
+		$metadata = array(
+			'file'             => __FILE__,
+			'name'             => 'unit-tests/test-block',
+			'viewScriptModule' => 'file:./blocks/notice/missing-asset.js',
+		);
+		$result   = register_block_script_module_id( $metadata, 'viewScriptModule' );
+
+		$this->assertSame( 'unit-tests-test-block-view-script-module', $result );
+	}
```

Comment:
@sirreal Why was this test needed? Why is `register_block_script_module_id()` not returning `false` here? I was attempting to harden the types for `WP_Script_Modules` by ensuring the `src` is always a `non-empty-string` but in this case what attempts to get registered is a `src` of `false` which results in an empty string.

Reply:
See https://github.com/WordPress/wordpress-develop/pull/9867/commits/4203864a1a90a106350cef25dc1ec91217e66541

Reply:
This is not fresh in my mind, forgive me.

I don't follow the question, what I see here appears to be a missing _asset_ file, which is allowed. The `src` should be derived from `file:./blocks/notice/missing-asset.js`. There's no corresponding `./blocks/notice/missing-asset.asset.php` file, which does not affect the ID produced by registering the module.

That's my interpretation of the test, but I have little recollection of this.